sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuJsonToStructs.scala

```diff
@@ -33,12 +33,37 @@ case class GpuJsonToStructs(
     timeZoneId: Option[String] = None)
     extends GpuUnaryExpression with TimeZoneAwareExpression with ExpectsInputTypes
         with NullIntolerant {
+
+  // Construct a dummy json string given an input schema, for example:
+  // schema = StructType([StructField("a", StringType), StructField("b", IntegerType)])
+  // returns "{"a": "", "b": 0}"
+  private def constructEmptyRow(): String = {
+    schema match {
+      case struct: StructType if (struct.fields.length > 0) => {
+        val res = struct.fields.foldRight ("") ((field, acc) => 
+          field.dataType match {
+            case IntegerType => "\"" + field.name + "\": 0, " + acc
+            case _ => "\"" + field.name + "\": \"\", " + acc
+          }
+        )
+        "{" + res.dropRight(2) + "}"
+      }
+      case _ => "{}"
+    }
+  }
   
-  private def cleanAndConcat(input: cudf.ColumnVector): (cudf.ColumnVector, cudf.ColumnVector) ={
-    withResource(cudf.Scalar.fromString("{}")) { emptyRow =>
-      val stripped = withResource(cudf.Scalar.fromString(" ")) { space =>
-        input.strip(space)
+  private def cleanAndConcat(input: cudf.ColumnVector): (cudf.ColumnVector, cudf.ColumnVector) = {
+    val emptyRowStr = constructEmptyRow()
+    withResource(cudf.Scalar.fromString(emptyRowStr)) { emptyRow =>
+
+      val stripped = if (input.getData == null) {
+        input.incRefCount
+      } else {
+        withResource(cudf.Scalar.fromString(" ")) { space =>
+          input.strip(space)
```

Comment:
Can we look for other places that strip, rstrip, and lstrip are called? I did a quick look and I think we might get the same problem on 

https://github.com/NVIDIA/spark-rapids/blob/e3b9517451e9bb8f7c024012a2e5c080540f4669/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOrcScan.scala#L386
https://github.com/NVIDIA/spark-rapids/blob/e3b9517451e9bb8f7c024012a2e5c080540f4669/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala#L306
https://github.com/NVIDIA/spark-rapids/blob/e3b9517451e9bb8f7c024012a2e5c080540f4669/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala#L286

But there may be others.

At a minimum we need to test these too and fix them if they are also broken

Reply:
I tested `GpuStringTrim[Left, Right]` implementation above. The same problem comes up. I added integration tests and fixes for them.

https://github.com/NVIDIA/spark-rapids/blob/e3b9517451e9bb8f7c024012a2e5c080540f4669/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOrcScan.scala#L386
I didn't find tests for the above `col.rstrip()` call site. I added a fix but I would need to confirm the fix is correct.

Reply:
I think that is fine. That is specifically for a special case ORC char type. We should file an issue for it just so it is not missed, but it should be really rare.